sympy/printing/tests/test_numpy.py

```diff
@@ -322,6 +323,10 @@ def test_issue_17006():
     N = MatrixSymbol("M", n, n)
     raises(NotImplementedError, lambda: lambdify(N, N + Identity(n)))
 
+def test_numpy_array():
+    assert NumPyPrinter().doprint(Array(((1, 2), (3, 5)))) == 'numpy.array([[1, 2], [3, 5]])'
```

Comment:
I wonder if we should add support (printer option? or per instance mapping?) for `dtype` right away. The array above will have its `dtype` inferred to be `int`, which (I suspect) could be surprising when used in e.g. division, what do you think?

Reply:
Not sure if I have any opinion really. I noted that `array` had a type argument, but sort of assumed/hoped that it was like in Matlab, complex double unless anything else is stated... But you are correct that it probably is better to be able to specify it.

Reply:
I aimed at doing the same thing as for matrices (which right now is a stupid statement since in this case I convert it to a Matrix and rely on the Matrix-printing, but if I would have generalized it...).

Reply:
And regarding extensions: not obvious that I will get around to it. No real in-depth knowledge of this part and just by accident tried out a few things yesterday that happened to work...